domain/schema/model/sql/0024-relation.sql

```diff
@@ -22,6 +22,25 @@ CREATE TABLE application_endpoint (
     REFERENCES charm_relation (uuid)
 );
 
+-- The application_endpoint ties an application's relation definition to an
+-- endpoint binding via a space. Only endpoint bindings which differ from the
+-- application default binding will be listed.
+CREATE TABLE application_extra_endpoint (
```

Comment:
Won't you ever need to join this table? I ask because in that case it's better to add a uuid now.

Reply:
It's a link table between an application and space and a charm extra binding. Unlike `application_endpoint` there is not other data related to this intersection.